gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/ConfigBasedDataset.java

```diff
@@ -89,9 +89,7 @@ private void calculateDatasetURN(){
       } catch (IOException e1) {
         // ignored
       }
-    }
-
-    this.datasetURN = e.toString();
+    } else this.datasetURN = e.toString();
```

Comment:
Addressed the comments on configuration name and moved the implementation to datasetFinder level. 
@ibuenros Could we first merge #1927 and #1876 since there's some interleaving with this PR? And I will update the PR afterwards.